### PR TITLE
Add policy rules for CycloneDX SBOM

### PR DESCRIPTION
**Pull request by lcarva**
Resolves: #746

### CODE REVIEW

File: policy/release/sbom_cyclonedx.rego [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	# https://cyclonedx.org/specification/overview/#recognized-predicate-type
+	statement.predicateType == "https://cyclonedx.org/bom"
+	sbom := statement.predicate
+]
+
+# _is_valid is true if the given CycloneDX SBOM has certain fields. This is not an exhaustive schema
+# check. It mostly ensures the fields used by the policy rules in this package have been set.
+_is_valid(sbom) if {
+	is_array(sbom.components)
```
Comment (zregvart) [inline policy/release/sbom_cyclonedx.rego:78]: Perhaps for future refinement we could use [`json.match_schema`](https://www.openpolicyagent.org/docs/latest/policy-reference/#builtin-object-jsonmatch_schema) here

**Review by zregvart — APPROVED** [review]

File: policy/release/sbom_cyclonedx.rego (hunk 1 above)
Comment (lcarva) [reply policy/release/sbom_cyclonedx.rego:78]: Great suggestion! I added some initial support but filed https://github.com/enterprise-contract/ec-policies/issues/808 as a follow up.

**Comment by lcarva** [comment]
> Very nice. Really clean and nice rego. Lgtm.
> 
> One thing I was wondering, should some of these rules be included in a collection, perhaps the redhat collection or perhaps minimal?

Yes! Added to both.

File: policy/release/sbom_cyclonedx_schema.rego
Comment (zregvart) [inline policy/release/sbom_cyclonedx_schema.rego:1]: :heart:

**Pull request merged**